### PR TITLE
Increase weeds size

### LINKED ISSUE #2075: weeds' range is smaller
image
image
New is first image, old is second.
This was never documented, as far as I can tell, in changelogs or anything I can find searching PRs on git. Therefore, I believe it to be an unintended change by @psykzz in #1375 .

### PR DESCRIPTION
**Pull request by psykzz**
Fixes #2075 

## About The Pull Request

Changes weed format from a star to a square.

## Why It's Good For The Game

This fixes a previously undocumented change, where the size was smaller..

## Changelog
:cl:
balance: Weed nodes are backing to creating squares of weeds, which should be much more similar to the original weeds before the last refactor
/:cl:

Before:
image

Current: 
image

New:

image

### CODE REVIEW

File: code/game/objects/effects/weeds.dm [hunk 1]
```diff
@@ -205,25 +205,55 @@
...
+	// 				// Finish here, but add it to expand weeds into
+	// 				node_turfs += AdjT
+	// 				continue
+
+	// 			turfs_to_check += AdjT
+	// 			node_turfs += AdjT
+
+	for(var/direction in GLOB.alldirs) //First pass to get the adjacent ones.
+		var/turf/adjacent_turf = get_step(src, direction)
+		if(adjacent_turf.density || LinkBlocked(src, adjacent_turf) || TurfBlockedNonWindow(adjacent_turf))
```
Comment (Rohesie) [inline code/game/objects/effects/weeds.dm]: This condition should produce a `continue` instead.

File: code/game/objects/effects/weeds.dm [hunk 2]
```diff
@@ -205,25 +205,55 @@
...
+							to_chat(world, "new turf - steraight")
+				if(NORTHEAST, NORTHWEST, SOUTHEAST, SOUTHWEST) //These are corner cases and check both their direction and their cardinal cases, three in total.
+					for(var/next_direction in list(expansion_dir, expansion_dir & NORTH|SOUTH, expansion_dir & WEST|EAST))
+						var/turf/next_turf = get_step(current_turf, next_direction)
+						if(!next_turf.density && !LinkBlocked(current_turf, next_turf) && !TurfBlockedNonWindow(next_turf) && !(next_turf in turfs_to_check))
+							turfs_to_check += next_turf
+							if(!(next_turf in node_turfs))
+								node_turfs += next_turf
+								to_chat(world, "new turf - diag")
+			turfs_to_check -= current_turf
```
Comment (Rohesie) [inline code/game/objects/effects/weeds.dm]: Not sure if this is nested in the last `for`, but it should be.

File: code/game/objects/effects/weeds.dm [hunk 3]
```diff
@@ -205,25 +205,55 @@
...
-				if (AdjT in node_turfs) // Ignore existing weeds
-					continue
-				if(AdjT.density || LinkBlocked(T, AdjT) || TurfBlockedNonWindow(AdjT))
-					// Finish here, but add it to expand weeds into
-					node_turfs += AdjT
-					continue
-
-				turfs_to_check += AdjT
-				node_turfs += AdjT
+	// var/list/turfs_to_check = list()
```
Comment (spookydonut) [inline code/game/objects/effects/weeds.dm]: delete commented out code.

File: code/game/objects/effects/weeds.dm (hunk 2 above)
Comment (spookydonut) [reply code/game/objects/effects/weeds.dm]: looks correct here

**Comment by Rohesie** [comment]
5x5 seems like a good size.
Given the weeds regrow, 7x7 is excessive.

**Comment by psykzz** [comment]
The original size was 7x7, but im happy to reduce it down if thats whats being asked?
The bug report was more about the sizing difference.

Im afraid that changing this down (even if it does regrow) will require a mapping change to ensure proper coverage of default weeds in maps.

**Comment by Rohesie** [comment]
Yes, it was 7x7 minus 4 corners, but it didn't regrow.
It's fine if the map is not fully weeded at roundstart,

**Comment by psykzz** [comment]
Updated weed size 

image

**Comment by psykzz** [comment]
This is technically fixed, but @Rohesie didnt like the performance impacts and wanted to take a stab at improving it. It could be merged knowing the performance issues, but thats a larger discussion.

**Comment by Rohesie** [comment]
My fix PR has grown stale.
This PR is fine by itself, but the feature could surely use some optimization. It's unnecessarily costly for what it does.

**Pull request merged**